src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/on_button_event_notification.cc

```diff
@@ -51,146 +52,37 @@ OnButtonEventNotification::OnButtonEventNotification(
-  ApplicationSharedPtr app;
-
-  // CUSTOM_BUTTON notification
-  if (static_cast<uint32_t>(mobile_apis::ButtonName::CUSTOM_BUTTON) == btn_id) {
-    // app_id is mandatory for CUSTOM_BUTTON notification
-    if (!is_app_id_exists) {
-      SDL_LOG_ERROR("CUSTOM_BUTTON OnButtonEvent without app_id.");
-      return;
-    }
-
-    app = application_manager_.application(
-        (*message_)[strings::msg_params][strings::app_id].asUInt());
-
-    // custom_button_id is mandatory for CUSTOM_BUTTON notification
-    if (false == (*message_)[strings::msg_params].keyExists(
-                     hmi_response::custom_button_id)) {
-      SDL_LOG_ERROR("CUSTOM_BUTTON OnButtonEvent without custom_button_id.");
-      return;
-    }
-
-    if (!app) {
-      SDL_LOG_ERROR("Application doesn't exist.");
-      return;
-    }
-
-    uint32_t custom_btn_id = 0;
-    custom_btn_id =
-        (*message_)[strings::msg_params][hmi_response::custom_button_id]
-            .asUInt();
-
-    if (false == app->IsSubscribedToSoftButton(custom_btn_id)) {
-      SDL_LOG_ERROR("Application doesn't subscribed to this custom_button_id.");
-      return;
-    }
-
-    const auto window_id = app->GetSoftButtonWindowID(custom_btn_id);
-    (*message_)[strings::msg_params][strings::window_id] = window_id;
-    const auto window_hmi_level = app->hmi_level(window_id);
-    if ((mobile_api::HMILevel::HMI_NONE == window_hmi_level)) {
-      SDL_LOG_WARN(
-          "CUSTOM_BUTTON OnButtonEvent notification is not allowed in "
-          "NONE hmi level");
-      return;
-    }
-
-    SendButtonEvent(app);
-    return;
-  }
+  auto& ref = (*message_)[strings::msg_params];
 
-  const std::vector<ApplicationSharedPtr>& subscribed_apps =
-      application_manager_.applications_by_button(btn_id);
-
-  std::vector<ApplicationSharedPtr>::const_iterator it =
-      subscribed_apps.begin();
-  for (; subscribed_apps.end() != it; ++it) {
-    ApplicationSharedPtr subscribed_app = *it;
-    if (!subscribed_app) {
-      SDL_LOG_WARN("Null pointer to subscribed app.");
-      continue;
-    }
-
-    // Send ButtonEvent notification only in HMI_FULL or HMI_LIMITED mode
-    const mobile_apis::HMILevel::eType app_hmi_level =
-        subscribed_app->hmi_level(
-            mobile_apis::PredefinedWindows::DEFAULT_WINDOW);
-    if ((mobile_api::HMILevel::HMI_FULL != app_hmi_level) &&
-        (mobile_api::HMILevel::HMI_LIMITED != app_hmi_level)) {
-      SDL_LOG_WARN("OnButtonEvent notification is allowed only"
-                   << "in FULL or LIMITED hmi level");
-      continue;
-    }
-    // if OK button and "app_id" absent send notification only in HMI_FULL mode
-    // otherwise send to subscribed apps in limited
-    if (is_app_id_exists || hmi_apis::Common_ButtonName::OK != btn_id ||
-        subscribed_app->IsFullscreen()) {
-      SendButtonEvent(subscribed_app);
-    }
-  }
-}
+  mobile_apis::ButtonEventMode::eType btn_event_mode;
+  if (ref.keyExists(hmi_response::button_mode)) {
```

Comment:
Is this created from both mobile and HMI requests? Do we need to check for both the HMI and mobile API parameter names?

Reply:
@iCollin I was able to see this request is created from HMI request only., so not sure if we need to check both APIs here. It looks mostly like a workaround for cases if it is created from the mobile side too. Please let me know if we need to remove redundant code.
For now, I moved this logic into `CreateButtonNotificationToMobile`